### PR TITLE
C#: Add test for `cs/dereferenced-value-may-be-null`

### PR DESCRIPTION
**Pull request by hvitved**
Record the FP reported [here](https://github.com/Semmle/ql/issues/2993).

### CODE REVIEW

File: csharp/ql/test/query-tests/Nullness/E.cs [hunk 1]
```diff
@@ -366,6 +366,16 @@ static void Ex33(string s, object o)
     static int Ex34(string s = null) => s.Length; // BAD (maybe)
 
     static int Ex35(string s = "null") => s.Length; // GOOD
+
+    static int Ex44(object o)
```
Comment (calumgrant) [inline csharp/ql/test/query-tests/Nullness/E.cs]: Why not `Ex36`?

Comment (hvitved) [reply csharp/ql/test/query-tests/Nullness/E.cs]: Now that is a very good question :-) I will change it.
